pkg/controller/build/build_controller_test.go

```diff
@@ -14,50 +16,54 @@ func Test_getBuildStrategyNamespace(t *testing.T) {
 	tests := []struct {
 		name string
 		args args
-		want string
+		want buildv1alpha1.StrategyRef
 	}{
 		{
-			name: "different namespace",
+			name: "namespaced scope buildstrategy",
 			args: args{
 				instance: &buildv1alpha1.Build{
 					ObjectMeta: metav1.ObjectMeta{
 						Name:      "build-sample",
 						Namespace: "build-ns",
 					},
 					Spec: buildv1alpha1.BuildSpec{
-						StrategyRef: metav1.ObjectMeta{
-							Name:      "my-strategy",
-							Namespace: "buildstrategy-ns",
+						StrategyRef: &buildv1alpha1.StrategyRef{
+							Name: "my-strategy",
+							Kind: "BuildStrategy",
 						},
 					},
 				},
 			},
-			want: "buildstrategy-ns",
+			want: buildv1alpha1.StrategyRef{
+				Name: "my-strategy",
+				Kind: "BuildStrategy",
+			},
 		},
 		{
-			name: "same namespace",
+			name: "cluster scope buildstrategy",
 			args: args{
 				instance: &buildv1alpha1.Build{
 					ObjectMeta: metav1.ObjectMeta{
 						Name:      "build-sample",
 						Namespace: "build-ns",
 					},
 					Spec: buildv1alpha1.BuildSpec{
-						StrategyRef: metav1.ObjectMeta{
-							Name:      "my-strategy",
-							Namespace: "build-ns",
+						StrategyRef: &buildv1alpha1.StrategyRef{
+							Name: "my-clusterstrategy",
+							Kind: "ClusterBuildStrategy",
 						},
 					},
 				},
 			},
-			want: "build-ns",
+			want: buildv1alpha1.StrategyRef{
+				Name: "my-clusterstrategy",
+				Kind: "ClusterBuildStrategy",
+			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := getBuildStrategyNamespace(tt.args.instance); got != tt.want {
```

Comment:
Curious, why are we removing this?

Reply:
Hi, 
In the PR, there are two BuildStrategy types:
1, get BuildStrategy from the same namespace of the build
2, get ClusterBuildStrategy from the cluster level

So there is no func `getBuildStrategyNamespace` anymore, we can get the namespace form `instance.namespace` directly.

And I think like Tekton Task and Cluster, we should not allow a build in namespace A which calls the buildstrategy in namespace B (across namespaces). If they want to call the buildstrategy out of his namespace, they should use the ClusterBuildStrategy instead.

Reply:
I understand, allowing a Build to access a namespace-scoped BuildStrategy from another namespace could be a security hole ( "escalation of privilege" using the service account ).

If we do support this in future, we could have a validating web-hook to add a rule using Subject access review checks:
 
```
User A can create a Build in namespace X
... using a BuildStrategy in Namespace Y
... IF User A has `view` privileges in Namespace Y
```

Reply:
Yes, I agree. If there is user requirement. We can add it in future.